internal/verifier/verifier.go

```diff
@@ -70,14 +61,23 @@ const (
 
 // Verifier is the object that verifies artifacts
 type Verifier struct {
-	verifier ArtifactVerifier
+	verifier verifyif.ArtifactVerifier
 	cacheDir string
 }
 
+// NewCustomVerifier creates a new Verifier object for testing using a custom verifier instance
+func NewCustomVerifier(verifier verifyif.ArtifactVerifier) *Verifier {
```

Comment:
I'm not sure why we need this wrapper; it seems like we should be passing around `ArtifactVerifier` instances...

Reply:
it's only for tests. I wasn't sure how to make an ingestor (which calls `NewVerifier` internally) make accept a version of a verifier that points to mocked `verifyif.ArtifactVerifier` instance without making the `verifier` field public to be able to set the instance there. So I came up with this one-off constructor that's really only used for tests which pass the mocked `verifyif.ArtifactVerifier` to it and out comes a `Verifier`.